### ISSUE TITLE
Rendering vertex corruption while replaying gfxr on Snapdragon 865 [SM8250]

### ISSUE THREAD

**Issue by niku2802** [LunarG/gfxreconstruct#507]
In this issue, vertices positions are getting corrupted while replaying gfxreconstruct however, this behavior is not seen with the actual app. When we connect renderdoc then also the issue is not seen while replaying the gfxreconstruct
There are in total 206 draw calls and from these 96th draw onward corruption is seen.

Attaching the corrupted and actual snap for just 96-97th draw with gfxr:
[gfxr.zip](https://github.com/LunarG/gfxreconstruct/files/5986181/gfxr.zip)

**Comment by mikes-lunarg** [comment]
Thanks for including the capture, I was able to reproduce the vertex corruption on a Snapdragon 845.

This appears to be an issue with the captured application. The vertex and index buffers are bound to a `VkDeviceMemory` object allocated with a `memoryTypeIndex` of 1, which on this hardware does not contain `VK_MEMORY_PROPERTY_HOST_COHERENT_BIT`. This sets up a race condition because the application does not call `vkFlushMappedMemoryRanges` after writing to the mapped memory, prior to `vkQueueSubmit`.  I have attached output from gfxrecon-toascii with additional debug output showing the memory types, memory ids and buffer ids: [406.txt](https://github.com/LunarG/gfxreconstruct/files/6085814/406.txt)

From the Vulkan spec:
> If a memory object does not have the VK_MEMORY_PROPERTY_HOST_COHERENT_BIT property, then vkFlushMappedMemoryRanges must be called in order to guarantee that writes to the memory object from the host are made available to the host domain, where they can be further made available to the device domain via a domain operation.

I believe you don't see this issue on the actual app because the timing is different. gfxrecon-replay does not preserve the original timing of API calls and this can sometimes expose underlying synchronization issues.
